src/handlers/commands.js

```diff
@@ -46,36 +46,101 @@ module.exports = async (message, gdb, db, countingChannel, prefix) => {
+module.exports.setupSlashCommands = async (client, db) => {
   client.ws.on("INTERACTION_CREATE", async interaction => {
     const commandFile = commands.get(interaciton.data.name);
     if (!commandFile) return respondWithError(interaction, `❌ The command \`${interaction.data.name}\` does not exist.`)
 
     const guild = client.guilds.cache.get(interaction_guild_id);
     if (!guild) return respondWithError(interaction, `❌ This guild is currently unavailable.`)
     const channel = guild.channels.cache.get(interaction.channel_id);
-    if (!channel) return respondWithError(interaction, `❌ I don't have access to view this channel.`)
+    if (!channel || !(commandFile.needAccessToChannel && !channel.viewable)) return respondWithError(interaction, `❌ I don't have access to view this channel.`)
+
+    const gdb = await db.guild(guild.id), { channel: countingChannel } = gdb.get();
+    if (channel.id == countingChannel && !commandFile.allowInCountingChannel) return respondWithError(interaction, "❌ This command is disabled inside the counting channel.")
+
     const member = guild.members.fetch(interaction.member.user.id);
 
-    
+    const permissionLevel = getPermissionLevel(member);
+    if (permissionLevel < commandFile.permissionRequired) return respondWithError(interaction, "❌ You don't have permission to do this.")
+
+    const args = {};
+    if (interaction.data.options) for (const slashArg of interaction.data.options) {
+      const { type, choices } = commandFile.options.find(o => o.name == slashArg.name);
+      args[slashArg.name] = convertArg(type, slashArg.value, choices, guild);
+    }
+
+    return commandFile.run(
+      {
+        send: async content => {
+          await client.api.interactions(interaction.id, interaction.token).callback.post({ data: { type: 4, data: { content } } });
+          return {
+            edit: content => client.api.webhooks(client.user.id, interaction.token).messages('@original').patch({ data: { content } }),
+            delete: () => client.api.webhooks(client.user.id, interaction.token).messages('@original').delete()
+          }
+        },
+        followup: async content => {
+          const m = await client.api.webhooks(client.user.id, interaction.token).post({ data: { content } });
+          return {
+            edit: content => client.api.webhooks(client.user.id, interaction.token).messages(m.id).patch({ data: { content } }),
+            delete: () => client.api.webhooks(client.user.id, interaction.token).messages(m.id).delete()
+          }
+        },
+        acknowledge: () => client.api.interactions(interaction.id, interaction.token).callback.post({ data: { type: 5 } }),
+      },
+      { client, guild, channel, member },
+      args, gdb,
+      { permissionLevel, db }
+    )
   })
 }
 
 const respondWithError = ({ id, token }, content) => client.api.interactions(id, token).callback.post({ data: { type: 3, data: { flags: 64, content }}})
```

Comment:
`client` is not defined

Reply:
Possible fix:
`client => ({ id, token }, content) => client.api.interactions(id, token).callback.post({ data: { type: 3, data: { flags: 64, content }}})`

and then set the `respondWithError` above in the `INTERACTION_CREATE` callback like `const respondWithError = respondWithErrorCallback(client)`